pkg/generate/app/cmd/newapp.go

```diff
@@ -325,10 +337,17 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if err != nil {
 					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
 				}
-				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses)
+				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses, app.PerfectMatchWeightedResolver{
```

Comment:
We should only need to create one resolver for the entire appconfig - the initializer up above that creates it should ensure this call gets the same resolver.